pkg/cmd/update.go

```diff
@@ -29,50 +35,85 @@ var updateCmd = &cobra.Command{
 	Short: "update the Doppler CLI",
 	Args:  cobra.NoArgs,
 	Run: func(cmd *cobra.Command, args []string) {
-		if installedViaBrew() {
-			brewUpdate()
-		} else {
-			utils.HandleError(fmt.Errorf("this command is not yet implemented for your install method"))
+		if utils.IsWindows() {
+			utils.HandleError(fmt.Errorf("this command is not yet implemented for your operating system"))
 		}
+
+		force := utils.GetBoolFlag(cmd, "force")
+		available, _, err := controllers.NewVersionAvailable()
+		if err != nil {
+			utils.HandleError(err)
+		}
+
+		if !available {
+			if force {
+				utils.Log(fmt.Sprintf("Already running the latest version but proceeding anyway due to --force flag"))
+			} else {
+				utils.Log(fmt.Sprintf("You are already running the latest version"))
+				return
+			}
+		}
+
+		runInstallScript()
 	},
 }
 
-func installedViaBrew() bool {
-	// this command returns 0 (nil) if installed, 1 (err) otherwise
-	command := []string{"brew", "ls", "--versions", "doppler"}
-	err := exec.Command(command[0], command[1:]...).Run() // #nosec G204
-	return err == nil
-}
+func runInstallScript() {
+	utils.Log("Updating...")
 
-func brewUpdate() {
-	utils.Log("Updating via Homebrew...")
-	command := []string{"brew", "upgrade", "--fetch-HEAD", "doppler"}
+	script, apiErr := http.GetCLIInstallScript()
+	if !apiErr.IsNil() {
+		utils.HandleError(apiErr.Unwrap(), apiErr.Message)
+	}
+
+	tmpFile, err := utils.WriteTempFile("install.sh", script, 0555)
+	// clean up temp file once we're done with it
+	defer os.Remove(tmpFile)
+
+	utils.LogDebug("Executing install script")
+	command := []string{tmpFile, "--debug"}
```

Comment:
Should we run this without --debug?

Reply:
We run with debug so that we can capture additional output. The output is only displayed when the user runs `doppler update` with the `--debug` flag.